PW_Autosplitter.asl

```diff
@@ -104,14 +101,58 @@ start
     (
         current.onMissionSequence == 1 &&
         old.onMissionSequence == 0 &&
-        current.onFreeMission == 0  && // the free mission flag may still be set when starting a campaign mission, but should flip to 0 after difficulty selection
+        current.onFreeMission == 0 && // the free mission flag may still be set when starting a campaign mission, but should flip to 0 after difficulty selection
         settings["Campaign"] == true
     );
 }
+
+init
+{
+    // Returns True when the mission Kings in the main campaign is complete.
+    // Rules consider this to be completion of the fadeout after Crimson 1.
+    vars.KingsSplit = (Func<byte, byte, byte, bool>)((currPhase, oldPhase, missionComplete) =>
+        {
+            return  currPhase == 7 && oldPhase == 6 && missionComplete == 2;
+        }
+    );
+
+    // Returns True when the mission Faust in Frontline 59 is complete.
+    // Rules consider this to be when the Frontline 59 logo cutscene starts.
+    vars.FaustSplit = (Func<bool>)(() =>
+        {
+            var controllerPawnPtr = new DeepPointer("ProjectWingman-Win64-Shipping.exe", 0x95AC140, 0x30, 0x250);
+            var pawnClassPtr = controllerPawnPtr.Deref<IntPtr>(game);
+            var pawnClass = game.ReadPointer(pawnClassPtr);
+            print("Pawn Class: " + pawnClass.ToString("X"));
+            return (pawnClass.ToString("X") == "7FF717B68550"); // During the logo cutscene, the WingmanPlayerController.Pawn is swapped from a FlyingPawn to a MF59Ending pawn
+        }
+    );
+
+    vars.GetLevelID = (Func<string>)(() =>
+        {
+        var levelPtr = new DeepPointer("ProjectWingman-Win64-Shipping.exe", 0x9150ED0, 0x0, 0x180, 0x490);
+        var fText = levelPtr.Deref<IntPtr>(game);
+
+        var data = game.ReadPointer(fText);
+        var length = game.ReadValue<short>(fText + 0x8);
+
+        return game.ReadString(fText, ReadStringType.UTF16, length);
+        });
```

Comment:
This function should be useful for https://github.com/Hilimii/ProjectWingman_Autosplitter/issues/10.